charts/portworx/templates/portworx-ds.yaml

```diff
@@ -82,18 +83,25 @@ spec:
           {{- with .Values }}
             [
               {{- if eq $drives "none" }}
-                {{- if eq $usedrivesAndPartitions true }}
-              "-A",
-                {{- else }}
-              "-a",
-                {{- end -}}
-                {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
-              "-f",
-                {{- end }}
-              {{- else }}
+                {{- if .AKSInstall -}}
```

Comment:
I don't think we should expose default drives to users. This should be in the template. For now we can assume there is only one default drive.
```
              {{- if eq $drives "none" }}
                {{- if .AKSInstall -}}
                   "-s", "type=Premium_LRS,size=150",
                {{- else -}}
                  {{- if eq $usedrivesAndPartitions true }}
                      "-A",
                  {{- else }}
                     "-a",
                  {{- end -}}
                 {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
                      "-f",
                  {{- end }}
                {{ end }}
              {{- else -}}
                {{- $driveNames := $drives | split ";" }}
                {{- range $index, $name := $driveNames }}
                  "-s", "{{ $name }}",
                {{- end -}}
              {{- end -}}
```

In future if we have defaults for EKS or other cloud providers, we can add another `else if` in that
```
              {{- if eq $drives "none" }}
                {{- if .AKSInstall -}}
                   "-s", "type=Premium_LRS,size=150",
                {{- else if .EKSInstall -}}
                   "-s", "type=gp2,size=150",
                {{- else -}}
                  {{- if eq $usedrivesAndPartitions true }}
                      "-A",
                  {{- else }}
                     "-a",
                  {{- end -}}
                 {{- if or $usefileSystemDrive $deployEnvironmentIKS }}
                      "-f",
                  {{- end }}
                {{ end }}
              {{- else -}}
                {{- $driveNames := $drives | split ";" }}
                {{- range $index, $name := $driveNames }}
                  "-s", "{{ $name }}",
                {{- end -}}
              {{- end -}}
```

Reply:
To add to this, we should also make sure, we have support to specify metadata drive if `internalKVDB: true`.